### PR TITLE
New default value for `DerivedQuantities.show_units`

### LINKED ISSUE #860: Deprecate `DerivedQuantities.show_units`
We need to remove the `show_units` argument of `DerivedQuantities`.

`show_units=True` will be the new behaviour.

### PR DESCRIPTION
**Pull request by RemDelaporteMathurin**
Fixes #860 but instead of deprecating the `show_units` argument, I just have it default to True.

What do you think @jhdark @KulaginVladimir? I thought it'd be nice to keep the functionality in case someone doesn't want the units in (eg. they are working in moles)...

### CODE REVIEW

**Review by jhdark — COMMENTED** [review]
Looks great, just a few minor comments from me. The largest change I think would be just to add this new export_unit property in each of the doc strings for the exports.

**Comment by jhdark** [comment]
Also might be worth adding another test for all the derived quantities without units, as many aren't being test at the moment (avg_surf, avg_vol, max_surf, max_vol, min_surf, min_vol, point_value, tot_surf, tot_vol). Just to get the full coverage

File: festim/exports/derived_quantities/derived_quantity.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def __init__(self, field) -> None:
         self.T = None
         self.data = []
         self.t = []
-        self.show_units = False
+        self.show_units = True
```
[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply festim/exports/derived_quantities/derived_quantity.py:42]: Well if it's not an argument then it's not really a default value. Like we don't document the fact that `data` has a "default" value of `[]` in the docstrings

File: festim/exports/derived_quantities/surface_flux.py [hunk 2]
```diff
@@ -53,15 +53,13 @@ def export_unit(self):
...
-        quantity_title = f"Flux surface {self.surface}: {self.field}"
-        if self.show_units:
-            if self.field == "T":
-                quantity_title = f"Heat flux surface {self.surface}"
-            else:
-                quantity_title = f"{self.field} flux surface {self.surface}"
+        if self.field == "T":
+            quantity_title = f"Heat flux surface {self.surface}"
+        else:
+            quantity_title = f"{self.field} flux surface {self.surface}"
```
[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply festim/exports/derived_quantities/surface_flux.py:59]: Well I wouldn't cause we may move away from "solute" at some point

File: test/unit/test_exports/test_derived_quantities/test_surface_flux.py [hunk 3]
```diff
@@ -20,7 +20,12 @@ def test_title(field, surface):
     """
 
     my_h_flux = SurfaceFlux(field, surface)
-    assert my_h_flux.title == "Flux surface {}: {}".format(surface, field)
+    my_h_flux.function = c_1D
+    if field == "T":
+        expected_title = f"Heat flux surface {surface} ({my_h_flux.export_unit})"
+    else:
+        expected_title = f"{field} flux surface {surface} ({my_h_flux.export_unit})"
```
[the remark this answers is not in the document]
Comment (RemDelaporteMathurin) [reply test/unit/test_exports/test_derived_quantities/test_surface_flux.py:27]: see previous comment

**Comment by RemDelaporteMathurin** [comment]
@jhdark @KulaginVladimir can we merge this?

**Review by KulaginVladimir — APPROVED** [review]
looks good 👍

**Pull request merged**